### PR TITLE
always have ban reason filled

### PR DESCRIPTION
**Pull request by Aprogiena**
this should make sure ban reason is filled, previously it was not filled anywhere (eh?)

### CODE REVIEW

**Comment by dangershony** [comment]
>previously it was not filled anywhere (eh?)  

🙃

File: src/Stratis.Bitcoin/Connection/PeerBanning.cs [hunk 1]
```diff
@@ -112,23 +112,24 @@ public PeerBanning(IConnectionManager connectionManager, ILoggerFactory loggerFa
...
+
             bool banPeer = true;
             NetworkPeer peer = this.connectionManager.ConnectedNodes.FindByEndpoint(endpoint);
             if (peer != null)
             {
                 ConnectionManagerBehavior peerBehavior = peer.Behavior<ConnectionManagerBehavior>();
                 if (!peerBehavior.Whitelisted)
                 {
-                    this.logger.LogDebug("Peer '{0}' banned for reason '{1}'.", endpoint, reason ?? "unknown");
+                    this.logger.LogDebug("Peer '{0}' banned for reason '{1}'.", endpoint, reason);
```
Comment (dangershony) [inline src/Stratis.Bitcoin/Connection/PeerBanning.cs]: This will not always be hit perhaps we should move this trace in to the if condition bellow?

Comment (Aprogiena) [reply src/Stratis.Bitcoin/Connection/PeerBanning.cs]: it looks fine here for me, where exactly do you mean to move it?

Comment (dangershony) [reply src/Stratis.Bitcoin/Connection/PeerBanning.cs]: If for what ever reason `NetworkPeer` is not in the collection we wont trace the ban.

I would put it here
```
 if (banPeer)
{		             
   this.logger.LogDebug("Peer '{0}' banned for reason '{1}'.", endpoint, reason);
   this.banStore.BanPeer(endpoint, this.dateTimeProvider.GetUtcNow().AddSeconds(banTimeSeconds));		                 this.banStore.BanPeer(endpoint, this.dateTimeProvider.GetUtcNow().AddSeconds(banTimeSeconds));
}		             
```

**Pull request merged**